### ISSUE TITLE
how to use dynamic shape of input_data

### ISSUE THREAD

**Issue by ywangeq** [tflearn/tflearn#970]
hello，i want to feed different of size of input to the network. And the tflearn has an interface call input_data to return the placeholder， however，I need to set the fixed size of shape when I build the network. Could anyone tell me some way to dynamically change the shape according to the real input size?
Thanks

**Comment by aymericdamien** [comment]
What kind of data are you feeding to the network? You can try to use padding / or dynamic calculation: see https://github.com/tflearn/tflearn/blob/master/examples/nlp/dynamic_lstm.py. In TensorFlow, the input size need to be fixed.

**Comment by ywangeq** [comment]
@aymericdamien Thanks for your response! Actually, I try to implement fcn with tflearn. Now I try to build the train. I  get the input and label by tensorflow where the type is tensor. But in the tflearn, the model.fit(X,Y) needs the array, and we know in tensorflow, we can directly use tensor.eval() to convert tensor back to array. Now how can i match these two things?
The code to get the img and label as follew:
 def read_images_from_disk(input_queue, input_size, random_scale, random_mirror, ignore_label, img_mean): 
    """
    Returns:
      Two tensors: the decoded image and its mask.
    """

    img_contents = tf.read_file(input_queue[0])
    label_contents = tf.read_file(input_queue[1])
    
    img = tf.image.decode_jpeg(img_contents, channels=3)
    img_r, img_g, img_b = tf.split(axis=2, num_or_size_splits=3, value=img)
    img = tf.cast(tf.concat(axis=2, values=[img_b, img_g, img_r]), dtype=tf.float32)
    # Extract mean.
    img -= img_mean

    label = tf.image.decode_png(label_contents, channels=1)

    if input_size is not None:
        h, w = input_size

        # Randomly scale the images and labels.
        if random_scale:
            img, label = image_scaling(img, label)

        # Randomly mirror the images and labels.
        if random_mirror:
            img, label = image_mirroring(img, label)

        # Randomly crops the images and labels.
        img, label = random_crop_and_pad_image_and_labels(img, label, h, w, ignore_label)

    return img, label

these two outputs can not feed to the model.fit(), maybe i have some misunderstanding here. Because the semantic segmentation's label is different with classification, so i can not use image_placeholder directly
